docs/basics/ipc.md

```diff
@@ -95,66 +97,154 @@ q)h({x+y};2;3) /Extend the list to pass more arguments
+```
+Using a seperate kdb+ instance, connect to the server and execute the functions:
+```q
+q)add:{x+2*y}         / define a function 'add' on the client
+q)h:hopen 5000        / connect to the server
+q)h(add;2;3)          / pass the client function 'add' to the server and execute, passing 2 parameters
+8
+q)h(`add;2;3)         / execute the 'add' function as defined on the server, passing 2 parameters
+5
 ```
 
-You may consider increasing the size of TCP send/receive buffers on your system to reduce the amount of blocking whilst trying to write into a socket.  
+!!! warning "Nesting sync requests is not recommended: response messages may be out of request order."
 
+#### One-shot message
 
-### Sync request (get)
+A sync message can also be sent on a short-lived connection. 
+When sending multiple messages, this is less efficient than [using a pre-existing connection](#sync-request-get) due to the effort of repeated connections/disconnections.
 
-Sends any pending outgoing (async) messages on `h`, sends the sync request message, and processes any pending incoming messages on `h` until a response (or error) message is received.
+A useful shorthand for a one-shot get is:
 
 ```q
-q)h"2+2" / this is sent to the remote process for calculation
-4
+q)`::5001 "1+1"
+2
 ```
 
-A useful shorthand for a one-shot get is:
+Since V4.0 2020.03.09, a one-shot query can be run with a timeout (in milliseconds), as in the second example below:
 
 ```q
-q)`::5001 "1+1"
+q)`::4567"2+2"
+4
+q)`::[(`::4567;100);"1+1"]
 2
 ```
 
-!!! warning "Nesting sync requests is not recommended: response messages may be out of request order."
+#### Interrupting requests
 
+It is possible to interrupt a long-running sync query with `kill -s INT *PID*`. As with the previous example, any subsequent attempt to communicate across this handle will fail.
+
+```q
+q)h"system\"sleep 30\""
+'rcv handle: 4. OS reports: Interrupted system call
+  [0]  h"system\"sleep 30\""
+       ^
+q)
+q)h"a"
+'Cannot write to handle 4. OS reports: Bad file descriptor
+  [0]  h"a"
+       ^
+```
 
 ### Response message (get response)
 
 Sent automatically by the listening process on completing a sync (get) request.
 
+### Async message (set)
 
-## Handle messages
+Serializes and puts a message on the output queue for handle `h`, and does not block client nor wait for any response message. A negative handle signifies async.
 
-Message handlers on the server are defined in the [`.z` namespace](../ref/dotz.md). Their default values can be overridden. The default handler for both sync and async messages is `value`:
+```q
+q)neg[h]"a:10" / on the remote instance, sets the variable a to 10
+```
+
+As per [sync messages](#sync-request-get), you can replace the string repesenting the code to execute as a list that forms a parse tree.
+
+Since the process is not waiting for a response, async querying is critical in situations where waiting for an unresponsive subscriber is unacceptable, e.g. in a tickerplant.
```

Comment:
```suggestion
Since the process is not waiting for a response, async querying is critical in situations where waiting for an unresponsive subscriber is unacceptable, for example, in a tickerplant.
```